### DISCUSSION TITLE
Minimal Storage Footprint of nnUNet trained model

### DISCUSSION THREAD

**Discussion by tjhendrickson** [MIC-DKFZ/nnUNet#1021] [General]
Hello,

I've trained a model with nnUNet and now would like to place it into containerized software so it can become a part of a larger pipeline. I was able to export it to a zip, but immediately noticed that the folder is quite large (1.2GB) to place into a container. Additional, trained models will likely be placed into the model, so I would like to minimize the storage footprint as much as possible. 

Here is what the folder looks like, along with various sub-folders and files. Can you pick anything out that could be excluded?
|-fold_0
 | |-debug.json
 | |-model_final_checkpoint.model
 | |-model_final_checkpoint.model.pkl
 | |-progress.png
 |-fold_1
 | |-debug.json
 | |-model_final_checkpoint.model
 | |-model_final_checkpoint.model.pkl
 | |-progress.png
 |-fold_2
 | |-debug.json
 | |-model_final_checkpoint.model
 | |-model_final_checkpoint.model.pkl
 | |-progress.png
 |-fold_3
 | |-debug.json
 | |-model_final_checkpoint.model
 | |-model_final_checkpoint.model.pkl
 | |-progress.png
 |-fold_4
 | |-debug.json
 | |-model_final_checkpoint.model
 | |-model_final_checkpoint.model.pkl
 | |-progress.png
 |-plans.pkl
 |-postprocessing.json

**Comment by FabianIsensee** [comment]
Hi, this looks as minimal as possible. Things you can do:
1) consider using only one model rather than the ensemble of 5 for inference. You need to train the 'all' fold for that (uses all training cases for training)
2) each checkpoint also contains the state dict of the optimizer, effectively doubling its size. You can remove that for inference but you may have to change the checkpoint loading code for things not to crash

Best,
Fabian
